### PR TITLE
Addition of new datasets

### PR DESCRIPTION
**Pull request by yeganehkordi**

### CODE REVIEW

File: encode_datasets.py [hunk 1]
```diff
@@ -1282,9 +1280,10 @@ def csqa2_process_test(file, dataset, kind):
         if '?' not in question:
             question = question + "?"
         id=questions[row][1]
+        answer=["-"]
 
         fmeta.write(f"{id} \n")
```
Comment (danyaljj) [inline encode_datasets.py:1285]: @yeganehkordi could you update the meta files for CSQA2.0 such that the 2nd column contains the index of correct answers? (represented as "A", "B", ..).

**Comment by danyaljj** [comment]
Inside the meta corresponding to "onestopqa_advanced" data I see the following: 
```
1	A
2	A
3	A
4	A
5	A
6	A
7	A
8	A
9	A
10	A
11	A
12	A
13	A
14	A
15	A
16	A
17	A
18	A
19	A
20	A
21	A
```
Is this correct? (all the answers are "A"?)

File: encode_datasets.py [hunk 2]
```diff
@@ -2059,7 +2064,228 @@ def process(inputfile, outfile):
...
+            answer_index = questions[row][7]
+            answer=chr(ord('A')+answer_index)
+            answer_string=questions[row][answer_index+3].strip().replace("\n", "").replace("\t", "").replace("   ", " ").replace("  ", " ")
+        fmeta.write(f"{id}\t{answer} \n")
+        fout.write(f"{question} \\n{options} \\n {contexts}\t{answer_string} \n")
+    return len(questions)
+
+def cosmosqa():
+    train_count=cosmosqa_process("train.csv","cosmos","train")
+    val_count=cosmosqa_process("valid.csv","cosmos","val")
```
Comment (danyaljj) [inline encode_datasets.py:2105]: "val" should be changed to "dev".

File: encode_datasets.py [hunk 3]
```diff
@@ -2059,7 +2064,228 @@ def process(inputfile, outfile):
...
+
+            answer=questions[row][5]
+            answer_index = ord(answer)-ord('A')
+            answer_string=questions[row][answer_index+1].strip().replace("\n", "").replace("\t", "").replace("   ", " ").replace("  ", " ")
+            fmeta.write(f"{counter}\t{answer} \n")
+            fout.write(f"{question} \\n{options} \t{answer_string} \n")
+    return counter
+
+def mmmlu():
+    val_count=mmmlu_process("measuring_massive_multitask_language_understanding","val")
```
Comment (danyaljj) [inline encode_datasets.py:2169]: "val" should be changed to "dev".

File: encode_datasets.py [hunk 4]
```diff
@@ -2059,7 +2064,228 @@ def process(inputfile, outfile):
...
+    else:
+        questions=df[['Question','Tweet','qid','Answer']].values
+
+    for row in range(len(questions)):
+        id=questions[row][2]
+        contexts=questions[row][1].strip().replace("\n", "").replace("\t", "").replace("   ", " ").replace("  ", " ")
+        question=questions[row][0].strip().replace("\n", "").replace("\t", "").replace("   ", " ").replace("  ", " ")
+        if '?' not in question:
+            question = question + "?"
+        fmeta.write(f"{id}\n")
```
Comment (danyaljj) [inline encode_datasets.py:2126]: Could you also dump all the correct answers here on the 2nd column of the meta-file? They could be separated with a special character like `///`.

File: encode_datasets.py (hunk 1 above)
Comment (yeganehkordi) [reply encode_datasets.py:1285]: Sure! Will do.

**Comment by yeganehkordi** [comment]
> Is this correct? (all the answers are "A"?)

Yes, all the answers were "A" in the dataset. Should I shuffle the options?

File: encode_datasets.py (hunk 2 above)
Comment (yeganehkordi) [reply encode_datasets.py:2105]: Done.

File: encode_datasets.py (hunk 3 above)
Comment (yeganehkordi) [reply encode_datasets.py:2169]: In this dataset, "val" and "dev" are two distinct splits. Should I merge them? As we don't have train split, we can change "val" to "train".

File: encode_datasets.py (hunk 4 above)
Comment (yeganehkordi) [reply encode_datasets.py:2126]: We have used `\t` for yes-no question meta-files. Is it okay to use it here as well?

File: encode_datasets.py [hunk 5]
```diff
@@ -1535,13 +1537,12 @@ def record_process_extractive (file,dataset, kind):
                     question = question + "."  
                 if kind is not "test":     
                     answer_string=json_line['qas'][index]['answers'][0]['text'].strip().replace("\n", "").replace("\t", "").replace("   ", " ").replace("  ", " ")
+                else:
+                    answer_string="-"
                 id=json_line['qas'][index]['idx']
 
                 fmeta.write(f"{id}\n")
```
Comment (danyaljj) [inline encode_datasets.py:1544]: Similarly here, can you output all the valid answers to the 2nd column of meta file, for `record_extractive` dataset?

**Pull request merged**